In [1]:
#@title Librerias
import numpy as np
import sympy as sp
from scipy.interpolate import CubicSpline
import matplotlib.pyplot as plt
from numpy.linalg import solve

# Actividad
Calcular el spline cúbico que interpola los puntos $(-1,5)$, $(0,3)$, $(1,3)$, $(2,4)$, encontrando los polinomios $S_0(x)$, $S_1(x)$ y $S_2(x)$. Evalue su respuesta para $x= 1.5$ usando `CubicSpline` de `scipy`.

In [3]:
x_val = 1.5

In [5]:
manual_polinomio = (64 + 18 * x_val + 6 * (x_val**2) - (x_val**3)) / 29
print(f'Método manual: {manual_polinomio:.3f}')

Método manual: 3.487


In [6]:
x, y = np.array([-1, 0, 1, 2]), np.array([5, 3, 3, 4])
cs = CubicSpline(x, y)
result = cs(x_val)
print(f'Método Scipy: {result.round(2)}')

Método Scipy: 3.44


Ambos métodos dieron un resultado alrededor de 3.4, esto confirma la confiabilidad del método de cubic splines.